sssom/io.py

```diff
@@ -361,3 +362,32 @@ def filter_file(input: str, output: TextIO, **kwargs) -> MappingSetDataFrame:
         if multiple_params and idx != len(params):
             query += " AND ("
     return run_sql_query(query=query, inputs=[input], output=output)
+
+
+def annotate_file(input: str, output: TextIO, **kwargs) -> MappingSetDataFrame:
+    """Annotate a file i.e. add custom metadata to the mapping set.
+
+    :param input: DataFrame to be queried over.
+    :param output: Output location.
+    :param **kwargs: Options provided by user
+        which are added to the metadata (e.g.: --mapping_set_id http://example.org/abcd)
+    :raises ValueError: If parameter provided is invalid.
+    :return: Annotated MappingSetDataFrame object.
+    """
+    params = {k: v for k, v in kwargs.items() if v}
+    legit_params = all(p in MAPPING_SET_SLOTS for p in params.keys())
+    if not legit_params:
```

Comment:
can we refactor this and all the rest to get an "augment_metadata()" method you can put into utils.py that takes as an input an msdf, and takes back an msdf where the metadata is augmented with the input, and has a sanitised parameter for the stuff that is supposed to be annotated (ideally just a "Metadata" object?) Something like

```
augment_metadata(msdf, meta: Metadata):
```
where meta is more or less your `params` object? this makes this method also usable by users of the python API.

Reply:
Done!